### DISCUSSION TITLE
Mandatory project-field in timesheet

### DISCUSSION THREAD

**Discussion by LinuxSquare** [kimai/kimai#2924] [General]
Hi there,
I'm fairly new to Kimai (discovered it like the day before yesterday and installed it yesterday on my server) and I'm liking so far the things I could peek with my eye. Unfortunately, there is something bothering me.

In the timesheet option, when I create a new timesheet for one of my customers, the field of the "Project"-option is mandatory, which is not very useful for my needs.
![image](https://user-images.githubusercontent.com/39629672/141357171-57e6f86a-62f8-49f6-bf34-94eccebeda97.png)

In my case, I would prefer it as an optional method, because sometimes it's a single time job, like going to a new customer and delivering his newly built PC. And in my opinion it's inefficient and when the time goes on, there would be more and more projects, just for building a PC for one of my customers, which would make the project-overview very confusing with all those "dead" projects.

Is there way, to make this field optional by modifying something in the configs? For me, a Project is like going for a longer period of time and not only a single-time job.

My system is running:
* OS: `Debian GNU/Linux 11 Bullseye`
* PHP-Version: `PHP 7.4 FPM`
* DB: `MariaDB 10.5.12`
* Webserver: `Nginx 1.18.0-6.1`

I'd appreciate any help I can get.

Thank you very much in advance.

Kind regards
LinuxSquare

**Comment by kevinpapst** [comment]
Understand your point and agree: I often find myself with the same problem. 

Anyways: a time record is connected through the project with a customer. 
So if the project would be optional, then there would be no customer connected to it.

So making the project optional is impossible without a huge refactoring of Kimai.

**Comment by LinuxSquare** [comment]
Is there any known workaround for example a default project that I can use for every of my customers e.g. `Single time job`?

**Comment by fsennj** [comment]
HI Kevin, great job on that project. i am also quite new and looking for a simple time tracking solution but without projects / customers.

So could this be done in future not to be mandatory (maybe invisible as project setting) for those with only one company and using only for internal time tracking

Or as Setting per User that tracks time to be set as default project and activity / customer so you dont need to fill it each time?

in the mobile app it is saved as favourite

thank you
Franz
